src/sentry/taskworker/client.py

```diff
@@ -142,6 +149,23 @@ def _get_cur_stub(self) -> tuple[str, ConsumerServiceStub]:
         if self._num_tasks_before_rebalance == 0:
             self._cur_host = random.choice(self._hosts)
             self._num_tasks_before_rebalance = self._max_tasks_before_rebalance
+            self._num_consecutive_unavailable_errors = 0
+            metrics.incr(
+                "taskworker.client.loadbalancer.rebalance",
+                tags={"reason": "max_tasks_reached"},
+            )
+        elif (
+            self._num_consecutive_unavailable_errors == self._max_consecutive_unavailable_errors
+            and len(self._hosts) > 1
+        ):
+            available_hosts = [h for h in self._hosts if h != self._cur_host]
```

Comment:
It would be good to remember which hosts have been marked as unavailable for a period of time. That would let us skip attempting to send results to that broker as we know the request will likely fail. Then when the time period has elapsed we resume sending results to the broker again.

Similarly it would be good if an unavailable broker was excluded from future load balancing choices for that period of time.